fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1062,6 +1055,315 @@ Future<Void> managementClusterRemoveTenantFromGroup(Transaction tr,
 	return Void();
 }
 
+template <class DB>
+struct RestoreClusterImpl {
+	MetaclusterOperationContext<DB> ctx;
+
+	// Initialization parameters
+	ClusterName clusterName;
+	ClusterConnectionString connectionString;
+	DataClusterEntry clusterEntry;
+	AddNewTenants addNewTenants;
+	RemoveMissingTenants removeMissingTenants;
+
+	// Variables to track state during restore
+	TenantName lastTenantName;
+	ClusterName lastClusterName;
+	uint64_t tenantId;
+	TenantName tenantName;
+	TenantName tenantNewName;
```

Comment:
I think most of this state can be tracked locally and passed via parameter to the functions you are calling based on the way they are used. It's a little bit confusing trying to keep track of this state when it frequently changes; usually I would recommend only storing things that are mostly unchanging and are globally applicable (e.g. the details of a tenant if you are running an operation on a single tenant).